rs/crypto/internal/crypto_service_provider/BUILD.bazel

```diff
@@ -2,7 +2,10 @@ load("@rules_rust//rust:defs.bzl", "rust_doc_test", "rust_library", "rust_test")
 load("//bazel:defs.bzl", "rust_test_suite_with_extra_srcs")
 load("//bazel:prost.bzl", "generated_files_check")
 
-package(default_visibility = ["//rs/crypto:__subpackages__"])
+package(default_visibility = [
+    "//rs/boundary_node/ic_boundary:__pkg__",
+    "//rs/crypto:__subpackages__",
```

Comment:
We definitely don't want this on the crypto side - in fact we're planning on removing this crate entirely, and anyway it's an internal detail that we don't want anything outside of `crypto` to depend on.

Can you expand on what you're trying to do here? An interface in the public crypto API may already exist, or we may be able to add something to suit.

Reply:
+1

@rikonor, is this related to [the discussion we had on Slack](https://dfinity.slack.com/archives/CH07MRJQ7/p1726846605820859) at some point? In particular, is the idea still the following:
>[...] the API BNs will generate a keypair, then register the public part with a canister. The canister will then store a mapping of node IDs to public keys, and this mapping can be referenced when one wants to encrypt data meant for a given API BN. The only caveat is that the private key will not be in the vault.

Just as a reminder: with the vetKeys feature, which will be available around end of Q1-2025/beginning of Q2-2025, you could do this directly without the need for custom key pairs.

IIUC, what you are trying to do is let the node sign an ingress message with it's signing key. You can see an example of how this can be done in the Orchestrator [here](https://github.com/dfinity/ic/blob/49000055d3fd5365860409bbcb68d6b884e9e097/rs/orchestrator/src/registration.rs#L410-L431). Basically, you have to instantiate a `CryptoComponent` and its `BasicSigner::sign_basic` implementation. Under the hood, this will use the _vault_, but you cannot use the vault directly.

An important factor, in any case, is that the respective process is allowed to talk to the vault. The Orchestrator process is explicitly allowed. Potentially, we have to allow the boundary node process as well, at least as long as we don't use vetKeys to do all of this.

Reply:
Hi @randombit and @fspreiss, thanks so much for your feedback and apologies about the delayed response, I just wanted to make sure I am able to make this change as you requested before replying - which should now be implemented. I switched to using `CryptoComponent` as you mentioned. Additionally, the CryptoConfig is now serialized and given to ic-boundary as a cli parameter.